egui/src/widgets/plot/legend.rs

```diff
@@ -1,81 +1,231 @@
-use std::string::String;
+use std::{
+    collections::{BTreeMap, HashSet},
+    string::String,
+};
 
 use crate::*;
 
-pub(crate) struct LegendEntry {
-    pub text: String,
-    pub color: Color32,
-    pub checked: bool,
-    pub hovered: bool,
+use super::items::PlotItem;
+
+/// Where to place the plot legend.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum LegendPosition {
+    TopLeft,
+    TopRight,
+    BottomLeft,
+    BottomRight,
```

Comment:
```suggestion
    LeftTop,
    RightTop,
    LeftBottom,
    RightBottom,
```

egui uses consistent X,Y order for everything. It reads a bit backwards, but one gets used to it :)

See for instance https://docs.rs/emath/0.12.0/emath/struct.Rect.html#method.left_top